transport-classes-kqueue/src/main/java/io/netty5/channel/kqueue/KQueueDatagramChannel.java

```diff
@@ -497,108 +497,99 @@ protected void doClose() throws Exception {
     @Override
     int readReady(ReadSink readSink) throws Exception {
         Buffer buffer = null;
-        int totalReadyBytes = 0;
         try {
             boolean connected = isConnected();
             boolean continueReading;
-            do {
-                buffer = readSink.allocateBuffer();
-                if (buffer == null) {
-                    readSink.processRead(0, 0, null);
-                    break;
-                }
-                assert buffer.isDirect();
-                int attemptedBytesRead = buffer.writableBytes();
-                int actualBytesRead = 0;
-                final DatagramPacket packet;
-                if (connected) {
-                    try {
-                        actualBytesRead = doReadBytes(buffer);
-                    } catch (Errors.NativeIoException e) {
-                        // We need to correctly translate connect errors to match NIO behaviour.
-                        if (e.expectedErr() == Errors.ERROR_ECONNREFUSED_NEGATIVE) {
-                            PortUnreachableException error = new PortUnreachableException(e.getMessage());
-                            error.initCause(e);
-                            throw error;
-                        }
-                        throw e;
+            buffer = readSink.allocateBuffer();
+            if (buffer == null) {
+                readSink.processRead(0, 0, null);
+                return 0;
+            }
+            assert buffer.isDirect();
+            int attemptedBytesRead = buffer.writableBytes();
+            int actualBytesRead = 0;
+            final DatagramPacket packet;
+            if (connected) {
+                try {
+                    actualBytesRead = doReadBytes(buffer);
+                } catch (Errors.NativeIoException e) {
+                    // We need to correctly translate connect errors to match NIO behaviour.
+                    if (e.expectedErr() == Errors.ERROR_ECONNREFUSED_NEGATIVE) {
+                        PortUnreachableException error = new PortUnreachableException(e.getMessage());
+                        error.initCause(e);
+                        throw error;
                     }
+                    throw e;
+                }
 
-                    if (actualBytesRead <= 0) {
-                        // nothing was read, release the buffer.
-                        buffer.close();
-
-                        readSink.processRead(attemptedBytesRead, actualBytesRead, null);
-                        // Signal closure
-                        if (actualBytesRead == -1) {
-                            totalReadyBytes = -1;
+                if (actualBytesRead <= 0) {
+                    // nothing was read, release the buffer.
+                    buffer.close();
+                    buffer = null;
```

Comment:
It looks redundant to set the buffer to null, since there's no finally-clause that will act on it.

Reply:
I think its not as we have a `cache Throwable` where we close the buffer if its not null.